### PR TITLE
Updates to brave libraries that are OSGi friendly

### PR DESCRIPTION
**Pull request by adriancole**
This moves to latest brave, which backports span v2 format to the old
api. It also corrects some OSGi and packaging glitches.

See #2151

### CODE REVIEW

**Comment by adriancole** [comment]
note: this is pending a release of brave, which will be out later today

**Comment by adriancole** [comment]
verified manually as well, using the example project

**Comment by oscerd** [comment]
LGTM :-) Thanks a lot. We'll wait the release :-)

**Comment by adriancole** [comment]
ask and ye shall receive.. looks like maven central is synced

**Comment by oscerd** [comment]
oh yeah, do you want to update the PR, passing from SNAPSHOT to the brand new release?

**Comment by adriancole** [comment]
heh, yeah I dropped the mic by accidentally trying to push to apache/camel
:P it should be bumped now

**Comment by oscerd** [comment]
Thanks a lot :-) I'm gonna merge :-)

File: platforms/karaf/features/src/main/resources/features.xml [hunk 1]
```diff
@@ -2416,10 +2416,10 @@
   <feature name='camel-zipkin' version='${project.version}' resolver='(obr)' start-level='50'>
     <details>camel-zipkin does not work in OSGi, see more at CAMEL-12088</details>
     <feature version='${project.version}'>camel-core</feature>
-    <bundle dependency='true'>mvn:io.zipkin.brave/brave/${brave-zipkin-version}</bundle>
     <bundle dependency='true'>mvn:io.zipkin.brave/brave-core/${brave-zipkin-version}</bundle>
     <bundle dependency='true'>mvn:io.zipkin.brave/brave-spancollector-scribe/${brave-zipkin-version}</bundle>
     <bundle dependency='true'>mvn:io.zipkin.java/zipkin/${zipkin-version}</bundle>
```
Comment (adriancole) [inline platforms/karaf/features/src/main/resources/features.xml]: oops I think this is unneeded now. Will remove

Comment (oscerd) [reply platforms/karaf/features/src/main/resources/features.xml]: Yes, I think so

Comment (adriancole) [reply platforms/karaf/features/src/main/resources/features.xml]: ok scrubbed

**Comment by oscerd** [comment]
Running the CamelZipkinTest in camel-itest-karaf I have

```shell
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 12.668 sec <<< FAILURE! - in org.apache.camel.itest.karaf.CamelZipkinTest
test(org.apache.camel.itest.karaf.CamelZipkinTest)  Time elapsed: 12.534 sec  <<< ERROR!
org.ops4j.pax.exam.WrappedTestContainerException: [test(org.apache.camel.itest.karaf.CamelZipkinTest): Unable to resolve root: missing requirement [root] osgi.identity; osgi.identity=camel-zipkin; type=karaf.feature; version="[2.21.0.SNAPSHOT,2.21.0.SNAPSHOT]"; filter:="(&(osgi.identity=camel-zipkin)(type=karaf.feature)(version>=2.21.0.SNAPSHOT)(version<=2.21.0.SNAPSHOT))" [caused by: Unable to resolve camel-zipkin/2.21.0.SNAPSHOT: missing requirement [camel-zipkin/2.21.0.SNAPSHOT] osgi.identity; osgi.identity=org.apache.camel.camel-zipkin; type=osgi.bundle; version="[2.21.0.SNAPSHOT,2.21.0.SNAPSHOT]"; resolution:=mandatory [caused by: Unable to resolve org.apache.camel.camel-zipkin/2.21.0.SNAPSHOT: missing requirement [org.apache.camel.camel-zipkin/2.21.0.SNAPSHOT] osgi.wiring.package; filter:="(&(osgi.wiring.package=com.github.kristofa.brave.internal)(version>=4.13.0)(!(version>=5.0.0))(braveinternal=true))" [caused by: Unable to resolve io.zipkin.brave.core/4.13.0: missing requirement [io.zipkin.brave.core/4.13.0] osgi.wiring.package; filter:="(&(osgi.wiring.package=zipkin)(version>=2.4.0)(!(version>=3.0.0)))"]]]]
	at org.apache.felix.resolver.ResolutionError.toException(ResolutionError.java:42)
	at org.apache.felix.resolver.ResolverImpl.doResolve(ResolverImpl.java:391)
	at org.apache.felix.resolver.ResolverImpl.resolve(ResolverImpl.java:377)
	at org.apache.felix.resolver.ResolverImpl.resolve(ResolverImpl.java:349)
	at org.apache.karaf.features.internal.region.SubsystemResolver.resolve(SubsystemResolver.java:218)
	at org.apache.karaf.features.internal.service.Deployer.deploy(Deployer.java:291)
	at org.apache.karaf.features.internal.service.FeaturesServiceImpl.doProvision(FeaturesServiceImpl.java:1248)
	at org.apache.karaf.features.internal.service.FeaturesServiceImpl.lambda$doProvisionInThread$1(FeaturesServiceImpl.java:1147)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

I love the OSGi's smell in the morning :-)

**Comment by adriancole** [comment]
bah drat

**Comment by adriancole** [comment]
ok I think I know what's up... plugging some osgi

**Comment by adriancole** [comment]
ok should be good for real this time. thanks for your help @oscerd!

**Comment by oscerd** [comment]
We just need to wait for maven central to be in sync. This is super-tested :-)

**Comment by adriancole** [comment]
should be good.. I downloaded from central prior to clicking..

Downloading from central:
https://repo.maven.apache.org/maven2/io/zipkin/brave/brave-core/4.13.1/brave-core-4.13.1.pom

Downloaded from central:
https://repo.maven.apache.org/maven2/io/zipkin/brave/brave-core/4.13.1/brave-core-4.13.1.pom
(4.0 kB at 2.6 kB/s)

Downloading from central:
https://repo.maven.apache.org/maven2/io/zipkin/brave/brave-parent/4.13.1/brave-parent-4.13.1.pom

Downloaded from central:
https://repo.maven.apache.org/maven2/io/zipkin/brave/brave-parent/4.13.1/brave-parent-4.13.1.pom
(28 kB at 31 kB/s)

Downloading from central:
https://repo.maven.apache.org/maven2/io/zipkin/brave/brave-spancollector-scribe/4.13.1/brave-spancollector-scribe-4.13.1.pom

Downloaded from central:
https://repo.maven.apache.org/maven2/io/zipkin/brave/brave-spancollector-scribe/4.13.1/brave-spancollector-scribe-4.13.1.pom
(2.0 kB at 3.4 kB/s)

Downloading from central:
https://repo.maven.apache.org/maven2/io/zipkin/brave/brave-core/4.13.1/brave-core-4.13.1.jar

Downloading from central:
https://repo.maven.apache.org/maven2/io/zipkin/brave/brave-spancollector-scribe/4.13.1/brave-spancollector-scribe-4.13.1.jar

Downloaded from central:
https://repo.maven.apache.org/maven2/io/zipkin/brave/brave-spancollector-scribe/4.13.1/brave-spancollector-scribe-4.13.1.jar
(58 kB at 67 kB/s)

Downloaded from central:
https://repo.maven.apache.org/maven2/io/zipkin/brave/brave-core/4.13.1/brave-core-4.13.1.jar
(152 kB at 86 kB/s)

**Comment by oscerd** [comment]
Ok, can you please just update the dependency in the parent/pom from SNAPSHOT to release? Thanks :-)

**Comment by adriancole** [comment]
second time I've had a push not push! force pushed for the win

**Comment by oscerd** [comment]
Thanks @adriancole . Finally merged on master :-) Closing the PR :-)

**Comment by adriancole** [comment]
thanks again, Andrea!

**Pull request closed without merging**